components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/listener/IdentityUserNameResolverListener.java

```diff
@@ -797,6 +804,36 @@ public boolean doPreGetUserListWithID(String claimUri, String claimValue, List<U
             }
         }
 
+        // Reflect newly removed users by listeners in returnUsersList
+        if (CollectionUtils.isNotEmpty(returnUserNamesList)) {
+            tempUserNamesList.addAll(returnInitialUserNamesList);
+            tempUserNamesList.removeAll(returnUserNamesList);
+            for (User user : returnUsersList) {
+                if (tempUserNamesList.contains(user.getUsername())) {
+                    returnUsersList.remove(user);
+                }
+            }
+            tempUserNamesList.clear();
+        }
+
+        // Reflect newly add users by listeners in returnUsersList
+        if (CollectionUtils.isNotEmpty(returnUserNamesList)) {
+            tempUserNamesList.addAll(returnUserNamesList);
+            tempUserNamesList.removeAll(returnInitialUserNamesList);
+            for (String username : tempUserNamesList) {
+                User newUser = new User();
+                newUser.setUsername(username);
+                try {
+                    newUser.setUserID(FrameworkUtils.resolveUserIdFromUsername(userStoreManager, username));
```

Comment:
**resolveUserIdFromUsername**, This method can be returned an empty user-id right? Don't we need to check its emptiness?

Reply:
This won't send empty id. It will create a new id if user doesn't have one.  Even if it is empty we don't need to handle it here as our main intension is to get the username

Reply:
ACK, that's why we don't throw the exception just logging.

Reply:
This won't send empty id. It will create a new id if user doesn't have one. Even if it is empty we don't need to handle it here as our main intension is to get the username